Formula/musepack.rb

```diff
@@ -31,9 +31,12 @@ class Musepack < Formula
   end
 
   def install
+    # Fix missing linkage to libm on Linux
+    (buildpath/"libmpcdec/CMakeLists.txt").append_lines "target_link_libraries(mpcdec m)\n" if OS.linux?
```

Comment:
has this been reported upstream?

Reply:
No. I guess @alebcay just found this out. We need to upstream this.

Reply:
Agreed. Just wanted to make sure this works. Not sure how this has compiled for so long without this (unless they only build/test on platforms where the math library is contained in `libc`).

I suspect the fix for upstream will be more than a one-liner though, as it would be better to check if linkage to `libm` is needed at all (or if `libm` even exists) rather than blindly adding a link library to the target.

Reply:
Do you know how to write that patch or should I help investigate?

Reply:
I'm working on writing the needed patch(es). They have both Autotools and CMake build systems available, so I'm trying to add the patch for both.

Reply:
My schedule has been busy so I don't think I will be able to help further with the patch for the time being. But I will provide here what I have so far. Adding this snippet to the end of `libmpcdec/CMakeLists.txt` should be a more well-rounded patch:
```CMake
include(CheckCSourceCompiles)
set(LIBM_TEST_SOURCE "#include<math.h>\ndouble x,y;int main(){log10(x);pow(x,y);return 0;}")

check_c_source_compiles("${LIBM_TEST_SOURCE}" HAVE_BUILTIN_MATH)
if(NOT HAVE_BUILTIN_MATH)
  set(CMAKE_REQUIRED_LIBRARIES m)
  check_c_source_compiles("${LIBM_TEST_SOURCE}" HAVE_LIBM_MATH)
  unset(CMAKE_REQUIRED_LIBRARIES)
  if(NOT HAVE_LIBM_MATH)
    message(FATAL_ERROR "Unable to find math library functions")
  endif()
  target_link_libraries(mpcdec m)
endif()
```
I tried to include as many of the math functions in `libmpcdec` as I could find in `LIBM_TEST_SOURCE`. But maybe it's overkill - or maybe there's more that should be added.

The Autotools workflow is trickier to patch because it seems there is math library usage in multiple subprojects, and the Autotools files do not work as-is. [`AC_SEARCH_LIBS`](https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Libraries.html) can be used in the `configure.in` file (which should be named `configure.ac` by convention) to attain this. Like in the CMake approach, I'm unsure if we should check just one of the math functions used, or all of them.